### PR TITLE
0.0.0.0/0 is not allowed

### PR DESCRIPTION
**Pull request by patel-bhavin**
this check now throws an error if the user has not changed the configuration

### CODE REVIEW

**Comment by patel-bhavin** [comment]
Right warning might be a better option! updated the PR

**Comment by peter-cg** [comment]
I doubt a warning is enough to bother people if it still defaults to open. How about changing `ip_whitelist = 0.0.0.0/0` to `# ip_whitelist = 1.2.3.4/32` in the default config?

**Comment by ghoto** [comment]
Agreed with @peter-cg. We also need to change the config, because a Warning goes lost with the verbosity that attack range already has.
Also, what about my PR regarding not having a default configuration, but a template configuration? So we can enforce the user to configure a few things, but also not disturbing their local changes to the config if they are working in a git branch of attack range?

**Comment by d1vious** [comment]
In case where a user wants to share their range with another, the 0.0.0.0/0 might be a valid option which is why I do not want to make it a hard error. @ghoto I like the idea of better addressing  it as a template config although we need to update docs to match this new workflow.

**Comment by ghoto** [comment]
@d1vious not saying throw hard error if IP is 0.0.0.0/24, I'm saying to force the user to explicitly type 0.0.0.0/24 or the addresses they want...

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,13 +0,0 @@
-repos:
-- repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (peter-cg) [inline .pre-commit-config.yaml:2]: Did you mean to delete this file?

**Review by P4T12ICK — APPROVED** [review]

**Pull request merged**